src/types/basic/uint.ts

```diff
@@ -128,6 +134,93 @@ export class NumberUintType extends UintType<number> {
   }
 }
 
+const TWO_POWER_32 = 2 ** 32;
+
+/**
+ * For 64 bit number, we want to operator on HashObject
+ * over bytes to improve performance.
+ */
+export class Number64UintType extends NumberUintType {
+  constructor() {
+    super({byteLength: 8});
+    this._typeSymbols.add(NUMBER_64_UINT_TYPE);
+  }
+
+  // TODO: move this logic all the way to persistent-merkle-tree?
+  struct_deserializeFromHashObject(data: HashObject, offset: number): number {
```

Comment:
Use chunkOffset or some name that clearly differentiates from regular byte offsets